### PR TITLE
Bw 535

### PR DESCRIPTION
**Pull request by milansar**

### CODE REVIEW

File: projects/trivia/src/app/components/app/app.component.tns.ts [hunk 1]
```diff
@@ -96,6 +99,47 @@ export class AppComponent implements OnInit, OnDestroy {
...
+        case '/recent-game':
+          this.utils.setScreenNameInFirebaseAnalytics(FirebaseScreenNameConstants.RECENT_COMPLETED_GAMES);
+          break;
+        case '/dashboard':
+          this.utils.setScreenNameInFirebaseAnalytics(FirebaseScreenNameConstants.DASHBOARD);
+          break;
+        case '/stats/leaderboard/':
+          this.utils.setScreenNameInFirebaseAnalytics(FirebaseScreenNameConstants.LEADERBOARD);
+          break;
+        case '/stats/leaderboard/':
```
Comment (milansar) [inline projects/trivia/src/app/components/app/app.component.tns.ts]: This is duplicate

**Comment by milansar** [comment]
@anihalaney , PR is ready for your review

File: projects/trivia/src/app/components/app/app.component.tns.ts [hunk 2]
```diff
@@ -96,6 +99,44 @@ export class AppComponent implements OnInit, OnDestroy {
...
     });
 
+    this.subscriptions.push(this.router.events.subscribe((evt) => {
+      if (!(evt instanceof NavigationEnd)) {
+        return;
+      }
+      switch (evt.urlAfterRedirects) {
+        case '/login':
+          this.utils.setScreenNameInFirebaseAnalytics(FirebaseScreenNameConstants.LOGIN);
+          break;
```
Comment (anihalaney) [inline projects/trivia/src/app/components/app/app.component.tns.ts]: should the component just dispatch an action and let the effect do the necessary analytics stuff?

**Comment by anihalaney** [comment]
@milansar - there is a lot of analytics logic in the component. I think it should be decoupled from the component and handled in effects. Let's talk about it.

**Pull request merged**